This only affects methods annotated with @SidedOnly? Because I'm pretty sure I have several cases of lambda's and method refs breaking badly in the server environment that don't involve @SidedOnly.

Everything works fine on the Client.

```
[18:37:58] [Server thread/FATAL] [minecraft/MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NoSuchMethodError: mods.railcraft.common.blocks.BlockEntityDelegate.getTileClass(Lnet/minecraft/block/state/IBlockState;)Ljava/lang/Class;
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_181]
	at net.minecraft.util.Util.runTask(Util.java:54) [Util.class:?]
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:798) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(DedicatedServer.java:415) [DedicatedServer.class:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:743) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:592) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
Caused by: java.lang.NoSuchMethodError: mods.railcraft.common.blocks.BlockEntityDelegate.getTileClass(Lnet/minecraft/block/state/IBlockState;)Ljava/lang/Class;
	at mods.railcraft.common.blocks.BlockEntityDelegate.lambda$18(BlockEntityDelegate.java:179) ~[BlockEntityDelegate.class:?]
	at mods.railcraft.common.blocks.TileManager.<init>(TileManager.java:44) ~[TileManager.class:?]
	at mods.railcraft.common.blocks.TileManager.forTile(TileManager.java:50) ~[TileManager.class:?]
	at mods.railcraft.common.blocks.BlockEntityDelegate.getCollisionBoundingBox(BlockEntityDelegate.java:179) ~[BlockEntityDelegate.class:?]
	at net.minecraft.block.state.BlockStateContainer$StateImplementation.getCollisionBoundingBox(BlockStateContainer.java:458) ~[BlockStateContainer$StateImplementation.class:?]
	at net.minecraft.world.World.mayPlace(World.java:3450) ~[World.class:?]
	at net.minecraft.item.ItemBlock.onItemUse(ItemBlock.java:52) ~[ItemBlock.class:?]
	at net.minecraftforge.common.ForgeHooks.onPlaceItemIntoWorld(ForgeHooks.java:876) ~[ForgeHooks.class:?]
	at net.minecraft.item.ItemStack.onItemUse(ItemStack.java:200) ~[ItemStack.class:?]
	at net.minecraft.server.management.PlayerInteractionManager.processRightClickBlock(PlayerInteractionManager.java:507) ~[PlayerInteractionManager.class:?]
	at net.minecraft.network.NetHandlerPlayServer.processTryUseItemOnBlock(NetHandlerPlayServer.java:769) ~[NetHandlerPlayServer.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.processPacket(CPacketPlayerTryUseItemOnBlock.java:68) ~[CPacketPlayerTryUseItemOnBlock.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.processPacket(CPacketPlayerTryUseItemOnBlock.java:13) ~[CPacketPlayerTryUseItemOnBlock.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(PacketThreadUtil.java:21) ~[PacketThreadUtil$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:1.8.0_181]
	at net.minecraft.util.Util.runTask(Util.java:53) ~[Util.class:?]
	... 5 more
```

--- a/src/main/java/net/minecraftforge/fml/common/asm/transformers/SideTransformer.java
+++ b/src/main/java/net/minecraftforge/fml/common/asm/transformers/SideTransformer.java
@@ -93,9 +93,10 @@ public class SideTransformer implements IClassTransformer
         }
 
         // remove dynamic synthetic lambda methods that are inside of removed methods
-        List<Handle> dynamicLambdaHandles = lambdaGatherer.getDynamicLambdaHandles();
-        if (!dynamicLambdaHandles.isEmpty())
+        for (List<Handle> dynamicLambdaHandles = lambdaGatherer.getDynamicLambdaHandles();
+             !dynamicLambdaHandles.isEmpty(); dynamicLambdaHandles = lambdaGatherer.getDynamicLambdaHandles())
         {
+            lambdaGatherer = new LambdaGatherer();
             methods = classNode.methods.iterator();
             while (methods.hasNext())
             {
@@ -110,6 +111,7 @@ public class SideTransformer implements IClassTransformer
                             System.out.println(String.format("Removing Method: %s.%s%s", classNode.name, method.name, method.desc));
                         }
                         methods.remove();
+                        lambdaGatherer.accept(method);
                     }
                 }
             }

